src/lib.rs

```diff
@@ -54,9 +54,9 @@ impl<T> Node<T> {
     /// Makes the given node come after this one, appropriately setting all other links.
     /// Assuming that self has a `next`.
     #[inline]
-    fn splice_next(&mut self, mut next: Box<Node<T>>) {
+    fn splice_next(&mut self, mut next: Box<Self>) {
         let mut old_next = self.next.take();
-        old_next.as_mut().map(|node| node.prev = Raw::some(&mut *next));
+        if let Some(ref mut node) = old_next { node.prev = Raw::some(&mut *next); }
```

Comment:
I'm not really convinced this is an improvement. I get some people are objectively opposed to side-effecting `map`, but I really hate if-let for most cases.